provider/core_dhcp_options_resource.go

```diff
@@ -154,6 +170,15 @@ func (s *DHCPOptionsResourceCrud) Get() (e error) {
 	res, e := s.Client.GetDHCPOptions(s.D.Id())
 	if e == nil {
 		s.Res = res
+
+		// If this is a default resource that we removed earlier, then
+		// we need to assume that the parent resource will remove it
+		// and notify terraform of it. Otherwise, terraform will
+		// see that the resource is still available and error out
+		if s.D.Get("manage_default_resource_id") != "" &&
```

Comment:
What happens if I have a default resource in my config, do a targeted delete, and then apply again? Does that work correctly?

Reply:
Yes, when you target delete a default resource, it is removed from the Terraform state but still exists in OCI. If you then try to add it to the config again, this will result in a Terraform Create request that's converted to an Update API call by the provider.